Add New MKDocs Documentation

rockenman1234 opened the pull request:
Hello all!

I've worked the last week on drafting a version of some docs for the Granite-IO API (to solve https://github.com/ibm-granite/granite-io/issues/138). This likely needs more work, but after a couple of hours of revisions on my end, I think it should be ready for someone else to take a look at.

I've included instructions on how to run the mkdocs server in the included `mkdocs/README.md` file.

Thanks for your time! 😃

rockenman1234 commented:
I updated the formatting and added some more eye candy! thx

frreiss commented on mkdocs/docs/design.md:
```diff
@@ -0,0 +1,128 @@
+# Granite IO Processing: Architecture & Design
```
Unless Github's change preview is deceiving me, it looks like this change set will leave us will two copies of `design.md`. Can you please update the contents of `granite_io/docs` so that there is only one copy of `design.md` and its images?

frreiss commented on mkdocs/README.md:
```diff
@@ -0,0 +1,47 @@
+# Granite-IO API Documentation with MKDocs
```
We have a few different ways of spelling the project name, and when we put all these files in the same place the differences become more noticeable. Can you please change the spelling to be `granite-io` across the board for consistency? That is, all lowercase, dash between words, and fixed-width font when possible.

frreiss commented on mkdocs/mkdocs.yml:
```diff
@@ -0,0 +1,35 @@
+site_name: IBM Granite-IO Documentation
+site_url: https://github.com/ibm-granite/granite-io
```
We'll probably start out by hosting these docs on Github Pages. From the docs at https://www.mkdocs.org/user-guide/deploying-your-docs/, it looks like the URL that comes from that process will be a URL under `github.io`.

frreiss reviewed (approved):
Looks good, some minor changes suggested inline.

rockenman1234 commented:
@frreiss I believe I've made the changes you requested! 

I went ahead and also fixed my unsigned commits (sorry that wasn't mentioned in the contrib guide), so please ignore the force-push.

I tweaked the new design file to better match the original, and removed the unneeded copy. Furthermore, I've updated the capitalization of the project name throughout the `/mkdocs` folder to now match `granite-io`. The site_url var in `/mkdocs/mkdocs.yml` is now `https://ibm-granite.github.io/granite-io/` instead of the previous `https://github.com/ibm-granite/granite-io`.

Please let me know what you think - and feel free to make any changes as needed! :-)
